pandas/indexes/base.py

```diff
@@ -2450,7 +2468,18 @@ def map(self, mapper):
 
         """
         from .multi import MultiIndex
-        mapped_values = self._arrmap(self.values, mapper)
+
+        if isinstance(mapper, ABCSeries):
```

Comment:
so I think its worth the effort to combine this code with ``Series.map`` (in reality you can prob just move the ``Series.map`` code to ``core.base.Base`` I think (with maybe some modifications here).